### ISSUE TITLE
Running "ChatterBox TTS" with the language set to Russian directly crashes ComfyUI.

### ISSUE THREAD

**Issue by 3272118850** [diodiogod/TTS-Audio-Suite#210]
My version：
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2025-12-23 19:28:00.033
** Platform: Windows
** Python version: 3.12.9 (tags/v3.12.9:fdb8142, Feb  4 2025, 15:27:58) [MSC v.1942 64 bit (AMD64)]
** Python executable: D:\ComfyUI\zuixin\python\python.exe
** ComfyUI Path: D:\ComfyUI\zuixin\ComfyUI
** ComfyUI Base Folder Path: D:\ComfyUI\zuixin\ComfyUI
** User directory: D:\ComfyUI\zuixin\ComfyUI\user
** ComfyUI-Manager config path: D:\ComfyUI\zuixin\ComfyUI\user\__manager\config.ini
** Log path: D:\ComfyUI\zuixin\ComfyUI\user\comfyui.log

Prestartup times for custom nodes:
   0.0 seconds: D:\ComfyUI\zuixin\ComfyUI\custom_nodes\ComfyUI-Easy-Use
   0.0 seconds: D:\ComfyUI\zuixin\ComfyUI\custom_nodes\rgthree-comfy
   2.6 seconds: D:\ComfyUI\zuixin\ComfyUI\custom_nodes\ComfyUI-Manager

Checkpoint files will always be loaded safely.
Total VRAM 32607 MB, total RAM 31849 MB
pytorch version: 2.7.0+cu128
xformers version: 0.0.30
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 5090 : cudaMallocAsync
Using async weight offloading with 2 streams
Enabled pinned memory 14332.0
Using xformers attention
Python version: 3.12.9 (tags/v3.12.9:fdb8142, Feb  4 2025, 15:27:58) [MSC v.1942 64 bit (AMD64)]
ComfyUI version: 0.5.1
ComfyUI frontend version: 1.34.9
[Prompt Server] web root: D:\ComfyUI\zuixin\python\Lib\site-packages\comfyui_frontend_package\static
Total VRAM 32607 MB, total RAM 31849 MB
pytorch version: 2.7.0+cu128
xformers version: 0.0.30
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 5090 : cudaMallocAsync
Using async weight offloading with 2 streams

——————————————

出现问题：

   🔇 Suppressed torchaudio 2.9 migration warnings
APEX FusedRMSNorm not available, using native implementation
ℹ️ Critical package versions: NumPy 2.1.3, Librosa 0.10.2, Numba 0.62.1, PyTorch 2.7.0+cu128, TorchAudio 2.7.0+cu128, Transformers 4.57.1, Accelerate 1.10.1, SoundFile 0.13.1
❌ F5-TTS not available: No module named 'flash_attn.bert_padding'
======================================================================
🚀 TTS Audio Suite v4.15.13
Universal multi-engine TTS extension for ComfyUI
✅ TTS Audio Suite v4.15.13 loaded with 29 nodes:
   • ⚙️ ChatterBox Official 23-Lang Engine
   • ⚙️ ChatterBox TTS Engine
   • ⚙️ F5 TTS Engine
   • ⚙️ Higgs Audio 2 Engine
   • ⚙️ IndexTTS-2 Engine
   • ⚙️ RVC Engine
   • ⚙️ Step Audio EditX Engine
   • ⚙️ VibeVoice Engine
   • 🌈 IndexTTS-2 Emotion Vectors
   • 🌈 IndexTTS-2 Text Emotion
   • 🌊 Audio Wave Analyzer
   • 🎙️ Voice Capture
   • 🎤 TTS Text
   • 🎨 Step Audio EditX - Audio Editor
   • 🎭 Character Voices
   • 🎭 Load RVC Character Model
   • 🏷️ Multiline TTS Tag Editor
   • 👄 F5-TTS Speech Editor
   • 📝 Phoneme Text Normalizer
   • 📺 TTS SRT
   • 🔄 Voice Changer
   • 🔧 Audio Analyzer Options
   • 🔧 F5-TTS Edit Options
   • 🔧 RVC Pitch Extraction Options
   • 🔧 Viseme Mouth Shape Options
   • 🗣️ Silent Speech Analyzer
   • 🤐 Noise or Vocal Removal
   • 🤐 Voice Fixer
   • 🥪 Merge Audio
======================================================================
WAS Node Suite: OpenCV Python FFMPEG support is enabled
WAS Node Suite Warning: `ffmpeg_bin_path` is not set in `D:\ComfyUI\zuixin\ComfyUI\custom_nodes\was-node-suite-comfyui\was_suite_config.json` config file. Will attempt to use system ffmpeg binaries if available.
[TTS Audio Suite] 🎭 Character voices: Found 26 characters, 10 aliases (cached)
[TTS Audio Suite] 🔄 Updating character discovery in background...
FETCH ComfyRegistry Data: 10/114
WAS Node Suite: Finished. Loaded 220 nodes successfully.

        "Art is the most intense mode of individualism that the world has known." - Oscar Wilde

——————————————

Error occurred when running the "⚙️ ChatterBox TTS Engine" workflow.：

got prompt
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
⚙️ ChatterBox Engine: Configured for Russian on auto
   Settings: exaggeration=0.5, temperature=0.8, cfg_weight=0.5
   Crash protection: hmm ,, {seg} hmm ,,
end_vram - start_vram: 8519680 - 8519680 = 0
#117 [ChatterBoxEngineNode]: 0.00s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#154 [VHS_LoadVideo]: 0.42s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#155 [AudioCrop]: 0.04s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#119 [PrimitiveStringMultiline]: 0.00s - vram 0b
🔍 Auto-detected language: english
end_vram - start_vram: 8519680 - 8519680 = 0
#118 [PhonemeTextNormalizer]: 0.00s - vram 0b
🎤 TTS Text: Using direct audio input (narrator)
⚠️ TTS Text: Direct audio input has no reference text - F5-TTS engines will fail
🎤 Generating Chatterbox for 'narrator' (lang: ru)
📎 Loading VoiceEncoder from English model (incomplete model fallback)
📁 Loading ve from local English model: D:\ComfyUI\zuixin\ComfyUI\models\TTS\chatterbox\English\ve.safetensors
⚠️ Failed to load local Russian model: Neither t3_cfg.safetensors nor t3_cfg.pt found in D:\ComfyUI\zuixin\ComfyUI\models\TTS\chatterbox\Russian
⚠️ Local model incomplete, missing: ['t3_cfg.safetensors', 's3gen.safetensors', 've.safetensors', 'conds.pt']
📥 Downloading Russian/t3_cfg.safetensors directly (no cache)

❌ Download failed for Russian/t3_cfg.safetensors: ('Connection aborted.', ConnectionResetError(10054, '远程主机强迫关闭了一个现有的连接。', None, 10054, None))
⚠️ Some files failed to download for Russian: ['t3_cfg.safetensors']
❌ Failed to download Russian model, falling back to English
📁 Using existing local model: D:\ComfyUI\zuixin\ComfyUI\models\TTS\chatterbox\English
input frame rate=25
🔤 Using language-specific tokenizer: tokenizer.json
📦 ChatterBox models loaded from: D:\ComfyUI\zuixin\ComfyUI\models\TTS\chatterbox\English
🔤 Final text to ChatterBox TTS model (narrator): 'Братцы, сейчас я сотрудничаю с казино ABEAR. Тут рил высокий шанс на занос. У меня на балансе уже куча денег =) Если вы мои ребята, залетайте в ABEAR, не пожалеете. Это лучшее казино из тех, что я пробовал когда-либо. Шансы лютые'
Sampling:  47%|████████████████████████████████▍                                    | 470/1000 [00:13<00:15, 35.12it/s]

D:\ComfyUI\zuixin\ComfyUI>

models

<img width="670" height="335" alt="Image" src="https://github.com/user-attachments/assets/107d7ca7-415f-433c-9526-84a1b57c58dd" />

**Comment by 3272118850** [comment]
Even after installing flash attention, the generation still fails, and the runtime error remains the same.

<img width="922" height="112" alt="Image" src="https://github.com/user-attachments/assets/786a3a60-4a29-48da-9c4e-1dd345ce8208" />

<img width="1106" height="573" alt="Image" src="https://github.com/user-attachments/assets/cc411af0-2d26-43bb-ba7d-2cd4793dcf40" />

**Comment by 3272118850** [comment]
> ## Fixed in v4.15.16
> The crash issue has been resolved. The Russian model download was failing because the files were located in a nested subdirectory that wasn't being accessed correctly.
> 
> ### What was fixed
> * Corrected the download path from  to `ru/``ru/chatterbox-ru-t3k/`
> * Russian model now downloads successfully (all 5 required files)
> * No more crashes during generation
> 
> ### Installation
> Update to v4.15.16 and the Russian model will download correctly when you use ChatterBox TTS Engine with Russian language.
> 
> ### Note about audio quality
> The community Russian model for classic ChatterBox () is an experimental fine-tune with no quality documentation. Some users report the output sounds unusual or English-like.`fron1runner/chatterbox-ru-t3k`
> 
> **Recommendation:** For better Russian TTS quality, use the **ChatterBox Official 23-Lang engine** instead, which has well-tested Russian support built in.
> 
> To switch:
> 
> 1. Use the  node instead of `⚙️ ChatterBox Official 23-Lang Engine``⚙️ ChatterBox TTS Engine`
> 2. Set language to Russian
> 3. Much better quality output
> 
> Let me know if you have any issues with v4.15.16!

Thank you for the update, author. I can download it now, but another issue has come up. After downloading the model, when I click to run it, it still crashes.

got prompt
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
[TimbreAudioLoader] 已加载 58 个音频文件
⚙️ ChatterBox Engine: Configured for local:Russian on auto
   Settings: exaggeration=0.5, temperature=0.8, cfg_weight=0.5
   Crash protection: hmm ,, {seg} hmm ,,
end_vram - start_vram: 8519680 - 8519680 = 0
#117 [ChatterBoxEngineNode]: 0.00s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#154 [VHS_LoadVideo]: 2.98s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#155 [AudioCrop]: 0.08s - vram 0b
end_vram - start_vram: 8519680 - 8519680 = 0
#119 [PrimitiveStringMultiline]: 0.00s - vram 0b
🔍 Auto-detected language: english
end_vram - start_vram: 8519680 - 8519680 = 0
#118 [PhonemeTextNormalizer]: 0.00s - vram 0b
🎤 TTS Text: Using direct audio input (narrator)
⚠️ TTS Text: Direct audio input has no reference text - F5-TTS engines will fail
🎤 Generating Chatterbox for 'narrator' (lang: local)
input frame rate=25
🔤 Using available tokenizer: tokenizer.json
📦 ChatterBox models loaded from: D:\ComfyUI\zuixin\ComfyUI\models\TTS\chatterbox\Russian
🔤 Final text to ChatterBox TTS model (narrator): 'Искусственный интеллект стремительно меняет нашу жизнь. Виртуальные ассистенты, умные дома, беспилотные автомобили — всё это становится частью повседневности. Технологии не только упрощают задачи, но и открывают новые возможности для творчества и обучения. Важно использовать их разумно, сохраняя человеческое взаимодействие и критическое мышление. Будущее уже наступает, и оно полно интересных вызовов.'

D:\ComfyUI\zuixin\ComfyUI>

<img width="2295" height="944" alt="Image" src="https://github.com/user-attachments/assets/afa633b3-58b5-4ac1-b0f3-1197270e43bd" />
This Russian text was generated using a large language model.

**Comment by diodiogod** [comment]
And I still think you should give up on this Russian model on classic chatterbox. I don't speak Russian, but it sounded really bad. Try the 23-lang chatterbox Engine.

**Comment by diodiogod** [comment]
I'm closing this for the lack of response, if you need we can reopen.
